src/test/blockmanager_tests.cpp

```diff
@@ -133,11 +132,70 @@ BOOST_FIXTURE_TEST_CASE(blockmanager_block_data_availability, TestChain100Setup)
     func_prune_blocks(last_pruned_block);
 
     // 3) The last block not pruned is in-between upper-block and the genesis block
-    BOOST_CHECK_EQUAL(&blockman.GetFirstBlock(tip, BLOCK_HAVE_DATA), first_available_block);
+    BOOST_CHECK_EQUAL(blockman.GetFirstBlock(tip, BLOCK_HAVE_DATA), first_available_block);
     BOOST_CHECK(blockman.CheckBlockDataAvailability(tip, *first_available_block));
     BOOST_CHECK(!blockman.CheckBlockDataAvailability(tip, *last_pruned_block));
 }
 
+BOOST_FIXTURE_TEST_CASE(blockmanager_block_data_part, TestChain100Setup)
+{
+    LOCK(::cs_main);
+    auto& chainman = m_node.chainman;
+    auto& blockman = chainman->m_blockman;
+    const CBlockIndex& tip = *chainman->ActiveTip();
+
+    std::vector<std::byte> block{};
+    BOOST_CHECK(blockman.ReadRawBlock(block, tip.GetBlockPos()));
+    BOOST_CHECK_GE(block.size(), 200);
+
+    std::vector<std::byte> block_part{};
+    BOOST_CHECK(blockman.ReadRawBlockPart(block_part, tip.GetBlockPos(), 0, std::nullopt));
```

Comment:
```suggestion
    const auto read_tip_part{[&](auto part_offset, auto part_size){return blockman.ReadRawBlockPart(block_part, tip.GetBlockPos(), part_offset, part_size);}};
    BOOST_CHECK(read_tip_part(0, std::nullopt));
```

style nit: Could reduce the verbosity  in this line (and below) with a lambda.